packages/Search/src/DTK_LinearBVH_decl.hpp

```diff
@@ -66,8 +65,37 @@ class BoundingVolumeHierarchy
   private:
     friend struct Details::TreeTraversal<DeviceType>;
 
-    Kokkos::View<Node *, DeviceType> _leaf_nodes;
-    Kokkos::View<Node *, DeviceType> _internal_nodes;
+    Kokkos::View<Node *, DeviceType> getInternalNodes()
+    { assert( !empty() ); return Kokkos::subview( _internal_and_leaf_nodes, std::make_pair(size_type{0}, size() - 1) ); }
+
+    Kokkos::View<Node *, DeviceType> getLeafNodes()
+    { assert( !empty() ); return Kokkos::subview( _internal_and_leaf_nodes, std::make_pair(size() - 1, 2 * size() - 1) ); }
+
+
+    KOKKOS_INLINE_FUNCTION
+    Node *getRoot() const { return _internal_and_leaf_nodes.data(); }
+
+    // FIXME not proud of that one
+    KOKKOS_INLINE_FUNCTION
+    bounding_volume_type &getBoundingVolume( Node const *node) const { return const_cast<bounding_volume_type&>(node->bounding_box); }
+
+    KOKKOS_INLINE_FUNCTION
+    size_t getLeafPermutationIndex( Node const *leaf ) const
```

Comment:
I don't really like this function as it depends on the state machine status of the whole thing. What I mean is that it is only valid to call it during tree traversal, but not before. But I don't have a good idea.